commerce-inventory-service/src/main/java/com/liferay/commerce/inventory/service/impl/CommerceInventoryWarehouseItemLocalServiceImpl.java

```diff
@@ -56,24 +57,33 @@ public CommerceInventoryWarehouseItem addCommerceInventoryWarehouseItem(
 			externalReferenceCode = null;
 		}
 
+		CommerceInventoryWarehouseItem commerceInventoryWarehouseItem =
+			commerceInventoryWarehouseItemPersistence.findByC_S(
+				commerceInventoryWarehouseId, sku);
+
+		if (commerceInventoryWarehouseItem != null) {
+			throw new DuplicateCommerceInventoryWarehouseItemException(
+				"Duplicated sku code");
+		}
+
 		long commerceInventoryWarehouseItemId = counterLocalService.increment();
 
-		CommerceInventoryWarehouseItem commerceInventoryWarehouseItem =
+		CommerceInventoryWarehouseItem commerceInventoryWarehouseItem2 =
```

Comment:
It makes more sense to me to reuse the `CommerceInventoryWarehouseItem commerceInventoryWarehouseItem` variable rather than declaring a new one here. Otherwise we should follow the [Where There's a 2, There's a 1](https://docs.google.com/document/d/188YFP0V4n9WJIPYZL3jGBRBvEsvBNmHOU97Fdd_g6Ew/edit#heading=h.urpkyx3jrv4y) rule.

Reply:
agree, @andrea-ale-sbarra can you fix?

Reply:
of course